live.ts

```diff
@@ -141,13 +153,38 @@ export const withLive = (
   };
 };
 
-export const live: () => Handlers<Page, WithLiveState> = () => ({
-  async GET(_, ctx) {
-    const page = await ctx.state.loadPage();
-    if (context.isDeploy && !page) {
+// Wrap any handler and it will receive `page` and `flags` along with any data on render.
+export const withLivePage = <Data>(
+  handler: Handler<Data & LivePageData, LiveState>,
+): Handler<Data, LiveState> => {
+  return async (req: Request, ctx) => {
+    await ctx.state.loadFlags();
+    await ctx.state.loadPage();
+    const freshRender = ctx.render;
+    ctx.render = (data?: Data) => {
+      const live = {
+        page: ctx.state.page,
+        flags: ctx.state.flags
+          ? ctx.state.flags.reduce((acc, flag) => {
+            acc[flag.key] = true;
+            return acc;
+          }, {} as Record<string, boolean>)
+          : {},
+      };
```

Comment:
Can you extract this into more intermediate steps and giving them names?

E.g: ` const flagsByKey = ctx.state.flags?.reduce(...`

Reply:
good idea